Assets/Scripts/UI/InGameUI/MenuController.cs

```diff
@@ -0,0 +1,147 @@
+{
+    public static MenuController Instance;
+
+    public Button buttonConstructor;
+    public Button buttonWorld;
+    public Button buttonWork;
+    public Button buttonOptions;
+    public Button buttonQuests;
+
+    private DialogBoxManager dialogBoxManager;
+
+    private MenuLeft menuLeft;
+
+    // Deactivates All Menus.
+    public void DeactivateAll()
+    {
+        menuLeft.CloseMenu();
+        WorldController.Instance.mouseController.ClearMouseMode(true);
+    }
+
+    // Toggles whether menu is active.
+    public void ToggleMenu(GameObject menu)
+    {
+        menu.SetActive(!menu.activeSelf);
+    }
+
+    public void OnButtonConstruction()
+    {
+        if (menuLeft.CurrentlyOpen != null && menuLeft.CurrentlyOpen.gameObject.name == "ConstructionMenu")
+        {
+            menuLeft.CloseMenu();
+        }
+        else
+        {
+            menuLeft.OpenMenu("ConstructionMenu");
+        }
+    }
+
+    public void OnButtonWork()
+    {
+        if (!GameController.Instance.IsModal)
+        {
+            DeactivateAll();
+            dialogBoxManager.dialogBoxJobList.ShowDialog();
+        }
+    }
+
+    public void OnButtonWorld()
+    {
+        if (!GameController.Instance.IsModal)
+        {
+            DeactivateAll();
+        }
+    }
+
+    public void OnButtonQuests()
+    {
+        if (!GameController.Instance.IsModal)
+        {
+            DeactivateAll();
+            dialogBoxManager.dialogBoxQuests.ShowDialog();
+        }
+    }
+
+    public void OnButtonOptions()
+    {
+        if (!GameController.Instance.IsModal)
+        {
+            DeactivateAll();
+            if (dialogBoxManager.dialogBoxSettings.isActiveAndEnabled)
+            {
+                dialogBoxManager.dialogBoxSettings.CloseDialog();
+            }
+
+            dialogBoxManager.dialogBoxOptions.ShowDialog();
+        }
+    }
+
+    // Use this for initialization.
+    private void Start()
+    {
+        dialogBoxManager = GameObject.Find("Dialog Boxes").GetComponent<DialogBoxManager>();
+        menuLeft = GameObject.Find("MenuLeft").GetComponent<MenuLeft>();
+
+        // Add listeners here.
+        buttonConstructor.onClick.AddListener(delegate
+        {
+            OnButtonConstruction();
+        });
+
+        buttonWorld.onClick.AddListener(delegate
+        {
+            OnButtonWorld();
+        });
+
+        buttonWork.onClick.AddListener(delegate
+        {
+            OnButtonWork();
+        });
+
+        buttonOptions.onClick.AddListener(delegate
+        {
+            OnButtonOptions();
+        });
+
+        buttonQuests = CreateButton("menu_quests");
+        buttonQuests.onClick.AddListener(delegate
+        {
+            OnButtonQuests();
+        });
+
+        DeactivateAll();
+    }
+
+    private Button CreateButton(string text)
+    {
+        GameObject buttonQuestGameObject = (GameObject)Instantiate(Resources.Load("UI/MenuButton"), this.gameObject.transform);
```

Comment:
You should change the name of the variable.
Is it a good idea to start having UI utilities for functions like this one?